responsibleai/model_monitor/components/src/calculate_attribution.py

```diff
@@ -0,0 +1,192 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+"""This file contains the core logic for feature attribution drift component."""
+import numpy as np
+import pandas as pd
+import logging
+
+from responsibleai import RAIInsights
+from sklearn.metrics import ndcg_score
+from ml_wrappers.model.predictions_wrapper import (
+    PredictionsModelWrapperClassification,
+    PredictionsModelWrapperRegression)
+from azureml.exceptions import UserErrorException
+
+try:
+    from lightgbm import LGBMClassifier, LGBMRegressor
+except ImportError:
+    pass
+
+from tabular.components.src._telemetry._loggerfactory import _LoggerFactory
+
+_logger = logging.getLogger(__file__)
+_ai_logger = None
+
+
+def _get_logger():
+    global _ai_logger
+    if _ai_logger is None:
+        _ai_logger = _LoggerFactory.get_logger(__file__)
+    return _ai_logger
+
+
+_get_logger()
+
+
+def create_lightgbm_model(X, y, task_type):
+    """Create model on which to calculate feature importances
+
+      :param X x values (data excluding target columns)
+      :type X pandas.Dataframe
+      :param y nparray of y values (target column data)
+      :type y nparray
+      :param task_type the task type (regression or classification) of the resulting model
+      :type task_type string
+      :return an appropriate model wrapper
+      :rtype: LightGBMClassifier or LightGBMRegressor
+    """
+    if (task_type == "classification"):
+        lgbm = LGBMClassifier(boosting_type='gbdt', learning_rate=0.1,
+                              max_depth=5, n_estimators=200, n_jobs=1, random_state=777)
+    else:
+        lgbm = LGBMRegressor(boosting_type='gbdt', learning_rate=0.1,
+                             max_depth=5, n_estimators=200, n_jobs=1, random_state=777)
+    model = lgbm.fit(X, y)
+
+    _logger.info("Created lightgbm model using task_type: {0}".format(task_type))
+    return model
+
+
+def get_model_wrapper(task_type, target_column, baseline_dataframe, production_dataframe):
+    """Create model wrapper using ml-wrappers on which to calculate feature importances
+
+      :param task_type The task type (regression or classification) of the resulting model
+      :type task_type string
+      :param target_column the column to predict
+      :type target_column string
+      :param baseline_dataframe The baseline data meaning the data used to create the
+      model monitor
+      :type baseline_dataframe pandas.DataFrame
```

Comment:
Missing semicolon after baseline_dataframe. In general, after the variable name there is a ":".